src/ol/expr/gpu.js

```diff
@@ -436,6 +436,7 @@ ${ifBlocks}
     const compiledIndex = compile(index, NumberType, context);
     return `texture2D(${paletteName}, vec2((${compiledIndex} + 0.5) / ${numColors}.0, 0.5))`;
   },
+  [Ops.ToNumber]: createCompiler(([value]) => value),
```

Comment:
Ok so this is going to be more involved I'm afraid. A compiled string in GLSL is turned into a number using `getStringNumberEquivalent()`. So here, since `value` is the compiled argument, it will most likely be a arbitrary number which has nothing to the input string.

Essentially, the GPU compiler cannot do string manipulation, and that includes conversion.

I attempted to a solution in [this commit](https://github.com/openlayers/openlayers/pull/15413/commits/89fb58d7fe223bfa25c6c59c7166da87fe1a2572), but it's not done yet. I'm actively working on this topic so, so hopefully in the near future string manipulations will become easier in the GPU compiler.

Reply:
Can I skip the `gpu.js` part, add a TODO comment and just merge the `expression` and `cpu` ?

Reply:
@jahow would it be ok ? Also I added 2 questions :)

Reply:
> Can I skip the `gpu.js` part, add a TODO comment and just merge the `expression` and `cpu` ?

That seems reasonable, yes; string manipulations in the GPU compiler will come later